src/Consumer.php

```diff
@@ -84,7 +85,8 @@ public function daemon($connectionName, $queue, WorkerOptions $options): void
             function (AMQPMessage $message) use ($connection, $options, $connectionName, $queue): void {
                 $this->gotJob = true;
 
-                $job = new RabbitMQJob(
+                /** @var RabbitMQJob $job */
+                $job = app(RabbitMQJobFactory::class)->create(
```

Comment:
Considering performance, I would suggest resolving factory just once before calling basic_consume, instead of resolving it every time message is consumed.

Reply:
revised